numpy/lib/npyio.py

```diff
@@ -1600,6 +1600,8 @@ def genfromtxt(fname, dtype=float, comments='#', delimiter=None,
             except TypeError:
                 usecols = [usecols, ]
     nbcols = len(usecols or first_values)
+    if not nbcols and converters is not None:
+        nbcols = len(converters)
```

Comment:
Turns out this is too naive and does not work for `converters={1: lambda arg: arg}`. I am not sure how to figure out `nbcols` in this case.